[build] Bump Go to 1.14

remeh opened the pull request:
### What does this PR do?

Build all Agents with Go 1.14

### Describe your test plan

- Validate that the Agents are running properly
- `datadog-agent version` should returns Go 1.14.7

remeh committed:
[build] Bump Go to 1.14

KSerrania commented on releasenotes/notes/bump-go-to-1.14.7-e7ca99c97da374b5.yaml:
```diff
@@ -0,0 +1,3 @@
+other:
+  - |
+    The Datadog Agent, Logs Agent and the system-probe are now built with ``Go 1.14.7``.
```
In previous go version bumps, we used the following formulation: "All Agents binaries are now compiled with Go \`\`x.y.z\`\`", so we should use the same here (cf. discussions in https://github.com/DataDog/datadog-agent/pull/5167, https://github.com/DataDog/datadog-agent/pull/5804, https://github.com/DataDog/datadog-agent/pull/5815).

remeh replied:
Sure thing. I've updated it in the same time as the builders images.

remeh committed:
[build] update builders images and use a clearer release note entry.

remeh committed:
[system-probe] `%w` is only available in `fmt.Errorf`

Fix the `go vet` call. Error appeared while upgrading to Go 1.14 toolchain.

remeh committed:
[bootstrap] update staticcheck to latest version (for compatibility with Go 1.14)

remeh committed:
[bootstrap] use release-branch of 1.14 for the golang.org/x/tools packages.

remeh committed:
Merge remote-tracking branch 'origin/master' into remeh/go-1.14

p-lambert committed:
Fix compilation errors due to missing KBUILD_MODNAME definition

remeh committed:
Properly escape parameter used with clang to build the sysprobe.

remeh committed:
[gitlab] we can now run the unit tests in the builders with -race mode.

Since we've switched to Go 1.14, there is no issues running in `-race` mode anymore.

remeh committed:
Merge remote-tracking branch 'origin/master' into remeh/go-1.14

remeh committed:
kubelet: properly switch off one of the kubelet test while running on Windows / AppVeyor.

remeh committed:
test/kubelet: one of the string error returned by the Go runtime has changed.

Adapting the test to make it work properly with Go1.14.

nammn reviewed (approved):
lgtm

derekwbrown commented on .gitlab-ci.yml:
```diff
@@ -88,14 +88,14 @@ variables:
-  DATADOG_AGENT_SYSPROBE_BUILDIMAGES: v2653795-e59e3fd
+  DATADOG_AGENT_BUILDIMAGES: v2894686-d80c3ce
+  DATADOG_AGENT_BUILDERS: v2894668-736db00
+  DATADOG_AGENT_WINBUILDIMAGES: v2894686-d80c3ce
```
it's probably semantics, but this tag is from your branch, not the resulting merge to master.

derekwbrown reviewed (approved):
Other than the one comment, LGTM.

ofek reviewed (approved):
:shipit:

hush-hush commented on .gitlab-ci.yml:
```diff
@@ -267,30 +267,6 @@ linux_armhf_go_deps:
   mkdir vendor && tar xzf vendor.tar.gz -C vendor
   rm -f go-bin.tar.gz go-pkg.tar.gz vendor.tar.gz
 
-# run tests for windows-64
```
Why do we remove those ?

remeh replied:
They were here just because we were not able to run the tests on Windows with Go < 1.14 with the `-race` flag.
Now that we can, we don't need those anymore because they covered by the "regular" Windows tests.

hush-hush commented on tasks/system_probe.py:
```diff
@@ -298,6 +298,7 @@ def build_object_files(ctx, bundle_ebpf=False):
         '-D__KERNEL__',
         '-DCONFIG_64BIT',
         '-D__BPF_TRACING__',
+        '-DKBUILD_MODNAME="\\"foo\\""',
```
What is this for ?

remeh replied:
While building the system probe, the build was complaining of this missing var during the build of the sysprobe module since recent changes, this fixes it https://github.com/DataDog/datadog-agent/pull/6254/commits/528f52fd281c32852915d97d660b25b69d076577

hush-hush commented on tasks/winbuildscripts/unittests.ps1:
```diff
@@ -56,11 +56,8 @@ if($err -ne 0){
     [Environment]::Exit($err)
 }
 
-if ($Env:NEW_BUILDER -eq "true"){
```
Is this linked to the change in `.gitlab.yaml` ?

remeh replied:
It is, this change makes the pipeline use `-race` for every tests on Windows.
